proxy/src/main/java/com/wavefront/agent/listeners/tracing/TracePortUnificationHandler.java

```diff
@@ -204,4 +206,13 @@ private static String parseError(String message, @Nullable ChannelHandlerContext
     }
     return errMsg.toString();
   }
+
+  private boolean sample(Span object) {
+    if (sampler != null && sampler.sample(object.getName(),
```

Comment:
if not marked Nullable, dont do null check
also if you do null check then default to true and not false